milli/src/update/delete_documents.rs

```diff
@@ -566,4 +606,51 @@ mod tests {
 
         wtxn.commit().unwrap();
     }
+
+    #[test]
+    fn delete_documents_with_filterable_attributes() {
+        let path = tempfile::tempdir().unwrap();
+        let mut options = EnvOpenOptions::new();
+        options.map_size(10 * 1024 * 1024); // 10 MB
+        let index = Index::new(options, &path).unwrap();
+
+        let mut wtxn = index.write_txn().unwrap();
+        let mut builder = Settings::new(&mut wtxn, &index, 0);
+        builder.set_primary_key(S("docid"));
+        builder.set_filterable_fields(hashset! { S("label") });
+        builder.execute(|_, _| ()).unwrap();
+
+        let content = &br#"[
+            {"docid":"1_4","label":"sign"},
+            {"docid":"1_5","label":"letter"},
+            {"docid":"1_7","label":"abstract,cartoon,design,pattern"},
+            {"docid":"1_36","label":"drawing,painting,pattern"},
+            {"docid":"1_37","label":"art,drawing,outdoor"},
+            {"docid":"1_38","label":"aquarium,art,drawing"},
+            {"docid":"1_39","label":"abstract"},
+            {"docid":"1_40","label":"cartoon"},
+            {"docid":"1_41","label":"art,drawing"},
+            {"docid":"1_42","label":"art,pattern"},
+            {"docid":"1_43","label":"abstract,art,drawing,pattern"},
+            {"docid":"1_44","label":"drawing"},
+            {"docid":"1_45","label":"art"},
+            {"docid":"1_46","label":"abstract,colorfulness,pattern"},
+            {"docid":"1_47","label":"abstract,pattern"},
+            {"docid":"1_52","label":"abstract,cartoon"},
+            {"docid":"1_57","label":"abstract,drawing,pattern"},
+            {"docid":"1_58","label":"abstract,art,cartoon"},
+            {"docid":"1_68","label":"design"},
+            {"docid":"1_69","label":"geometry"}
+        ]"#[..];
+        let mut builder = IndexDocuments::new(&mut wtxn, &index, 0);
+        builder.update_format(UpdateFormat::Json);
+        builder.execute(content, |_, _| ()).unwrap();
+
+        // Delete not all of the documents but some of them.
+        let mut builder = DeleteDocuments::new(&mut wtxn, &index, 1).unwrap();
+        builder.delete_external_id("1_4");
+        builder.execute().unwrap();
+
+        wtxn.commit().unwrap();
```

Comment:
Why don't we check that the corresponding ID is deleted from the facet database or searching for the label `drawing` and checking if the document `1_4` is not present?